### PR TITLE
Resolve batch of Cursor PR comments

### PR DESCRIPTION
**Pull request by TechNickAI**
## Summary

Batch of fixes from resolving old Cursor PR comments across multiple areas:

- **Tool streaming** - Better tool call ID handling (prefer `toolCallId` over `id`), improved error detection from multiple formats (string prefix, error field, exit codes)
- **Jobs API** - Inline error responses with proper HTTP status codes, added Sentry context for Temporal failures, prevents DB/Temporal state inconsistency
- **UI components** - Button ref merging for ripple effect with `asChild`, StatefulButton respects `prefers-reduced-motion`
- **Evals** - Fixed OpenRouter model name format, removed examples section from title prompt
- **Security** - Symlink rejection in project path validation, email format validation, message pruning edge case when system messages exceed max

## Test plan

- [x] All 1769 tests pass
- [x] Type check passes
- [x] Lint/format passes

Generated with Carmenta

### CODE REVIEW

**Comment by TechNickAI** [comment]
Great catch on the `pruneModelMessages` consistency issue! Applied the same edge case fix. 

The button ref merging observation is fair - there's duplication between the asChild/non-asChild branches. I'll leave that as-is for now since it's working and not blocking, but noted as a cleanup opportunity.

File: evals/title/autoevals-scorers.ts [hunk 1]
```diff
@@ -136,7 +136,7 @@ Rate the title quality:`,
         // Poor: too generic, too long, question-formatted, or misses topic
         Poor: 0.25,
     },
-    model: "gpt-4o-mini",
+    model: "openai/gpt-4o-mini",
```
[the remark this answers is not in the document]
Comment (TechNickAI) [reply evals/title/autoevals-scorers.ts]: Good catch! Reverted to `gpt-4o-mini` which works with both OpenRouter and the OpenAI fallback path.

File: app/api/jobs/[jobId]/route.ts [hunk 2]
```diff
@@ -162,10 +188,32 @@ export async function PATCH(request: NextRequest, context: RouteContext) {
...
+
+                // Don't proceed with DB update - would create inconsistency
+                return NextResponse.json(
+                    {
+                        error: "Unable to update job",
+                        message:
+                            "Background processing is temporarily unavailable. Please try again in a few minutes.",
+                    },
+                    { status: 503 }
+                );
```
[the remark this answers is not in the document]
Comment (TechNickAI) [reply app/api/jobs/[jobId]/route.ts]: Valid edge case observation, but declining this one. The scenario requires a request that updates BOTH schedule AND pause state simultaneously, which is rare in practice (usually these are separate user actions).

The new code is strictly better than before:
- **Before**: Temporal op could fail silently, DB updates anyway → guaranteed DB/Temporal drift
- **After**: Temporal op failure returns 503 early → user knows it failed, can retry

A full fix would require distributed transaction semantics (saga/compensating actions) which is over-engineering for this edge case. The current behavior is an improvement, not perfect.
